pkg/planner/core/casetest/vectorsearch/vector_index_test.go

```diff
@@ -0,0 +1,186 @@
+
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t1")
+	tk.MustExec(`
+		create table t1 (
+			vec vector(3),
+			a int,
+			b int,
+			c vector(3),
+			d vector
+		)
+	`)
+	tk.MustExec("alter table t1 set tiflash replica 1;")
+	tk.MustExec("alter table t1 add vector index ((vec_cosine_distance(vec))) USING HNSW;")
+	tk.MustExec(`
+		insert into t1 values
+			('[1,1,1]', 1, 1, '[1,1,1]', '[1,1,1]'),
+			('[2,2,2]', 2, 2, '[2,2,2]', '[2,2,2]'),
+			('[3,3,3]', 3, 3, '[3,3,3]', '[3,3,3]')
+	`)
+	for i := 0; i < 14; i++ {
+		tk.MustExec("insert into t1(vec, a, b, c, d) select vec, a, b, c, d from t1")
+	}
+	tk.MustExec("analyze table t1")
+
+	tk.MustExec("set @@tidb_isolation_read_engines = 'tiflash'")
+
+	var input []string
+	var output []struct {
+		SQL  string
+		Plan []string
+		Warn []string
+	}
+	integrationSuiteData := GetANNIndexSuiteData()
+	integrationSuiteData.LoadTestCases(t, &input, &output)
+	for i, tt := range input {
+		testdata.OnRecord(func() {
+			output[i].SQL = tt
+			output[i].Plan = testdata.ConvertRowsToStrings(tk.MustQuery(tt).Rows())
+			output[i].Warn = testdata.ConvertSQLWarnToStrings(tk.Session().GetSessionVars().StmtCtx.GetWarnings())
+		})
+		tk.MustQuery(tt).Check(testkit.Rows(output[i].Plan...))
+		require.Equal(t, output[i].Warn, testdata.ConvertSQLWarnToStrings(tk.Session().GetSessionVars().StmtCtx.GetWarnings()))
+	}
+}
+
+func TestANNIndexNormalizedPlan(t *testing.T) {
+	store := testkit.CreateMockStoreWithSchemaLease(t, 1*time.Second, mockstore.WithMockTiFlash(2))
+
+	tk := testkit.NewTestKit(t, store)
+
+	tiflash := infosync.NewMockTiFlash()
+	infosync.SetMockTiFlash(tiflash)
+	defer func() {
+		tiflash.Lock()
+		tiflash.StatusServer.Close()
+		tiflash.Unlock()
+	}()
+
+	testfailpoint.Enable(t, "github.com/pingcap/tidb/pkg/ddl/MockCheckVectorIndexProcess", `return(1)`)
+
+	getNormalizedPlan := func() ([]string, string) {
+		info := tk.Session().ShowProcess()
+		require.NotNil(t, info)
+		p, ok := info.Plan.(base.Plan)
+		require.True(t, ok)
+		plan, digest := core.NormalizePlan(p)
+
+		// test the new normalization code
+		flat := core.FlattenPhysicalPlan(p, false)
+		newNormalized, newDigest := core.NormalizeFlatPlan(flat)
+		require.Equal(t, plan, newNormalized)
+		require.Equal(t, digest, newDigest)
+
+		normalizedPlan, err := plancodec.DecodeNormalizedPlan(plan)
+		normalizedPlanRows := getPlanRows(normalizedPlan)
+		require.NoError(t, err)
+
+		return normalizedPlanRows, digest.String()
+	}
+
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec(`
+		create table t (
+			vec vector(3)
+		)
+	`)
+	tk.MustExec("alter table t set tiflash replica 1;")
+	tk.MustExec("alter table t add vector index ((vec_cosine_distance(vec))) using hnsw;")
+	tk.MustExec(`
+		insert into t values
+			('[1,1,1]'),
+			('[2,2,2]'),
+			('[3,3,3]')
+	`)
+
+	tk.MustExec("analyze table t")
+
+	tk.MustExec("set @@tidb_isolation_read_engines = 'tiflash'")
```

Comment:
Could we add `tikv` store type in this test like https://github.com/tidbcloud/tidb-cse/pull/1131/commits/50c1db52d8b11eeecbb378ce27d872ac1066541d#diff-deeb61ba363f7beece976930a7bf135418eecab893115f14e339213fd845aa4aR144?